tests/unit/overlays/test_overlay_manager.py

```diff
@@ -150,3 +150,184 @@ def test_mkdirs(self, new_dir):
         Path("overlay/overlay").is_dir()
         Path("overlay/packages").is_dir()
         Path("overlay/work").is_dir()
+
+
+class TestPackageManagement:
+    """Verify package installation on mounted overlayfs."""
+
+    @pytest.fixture(autouse=True)
+    def setup_method_fixture(self, new_dir):
+        # pylint: disable=attribute-defined-outside-init
+        info = ProjectInfo(application_name="test", cache_dir=new_dir)
+        self.p1 = Part("p1", {"plugin": "nil"})
+        self.p2 = Part("p2", {"plugin": "nil"})
+        base_layer_dir = Path("base_dir")
+        base_layer_dir.mkdir()
+        self.om = OverlayManager(
+            project_info=info,
+            part_list=[self.p1, self.p2],
+            base_layer_dir=base_layer_dir,
+        )
+        # pylint: enable=attribute-defined-outside-init
+
+    def test_refresh_packages_list(self, mocker, new_dir):
+        mock_mount = mocker.patch("craft_parts.utils.os_utils.mount")
+        mock_chroot = mocker.patch("pychroot.Chroot")
+        mock_refresh_packages_list = mocker.patch(
+            "craft_parts.packages.Repository.refresh_packages_list"
+        )
```

Comment:
can't these all be patched in the setup_method_fixture? If refresh is patched when install is to be used shouldn't matter much and at least you don't accidentally trigger the wrong thing and break your env :-)

Reply:
Moved common overlay manager test mocks to test class setup.